File: code/typescript/strategy/StrategyMain.ts
import { RoutePlanner } from "./RoutePlanner";
import { FastestRouteStrategy } from "./FastestRouteStrategy";
import { ShortestRouteStrategy } from "./ShortestRouteStrategy";

export function Strategy(): void {
    const planner = new RoutePlanner();

    // Set the strategy to calculate the fastest route
    planner.setStrategy(new FastestRouteStrategy());
    planner.executeStrategy();

    // Set the strategy to calculate the shortest route
    planner.setStrategy(new ShortestRouteStrategy());
    planner.executeStrategy();
}

// Invoke Strategy function
Strategy();

File: code/csharp/DesignPatterns/strategy/StrategyMain.cs
using System;

namespace Strategy
{
    public class StrategyMain
    {
        public static void Strategy()
        {
            RoutePlanner planner = new RoutePlanner();

            // Set the strategy to calculate the fastest route
            planner.SetStrategy(new FastestRouteStrategy());
            planner.ExecuteStrategy();

            // Set the strategy to calculate the shortest route
            planner.SetStrategy(new ShortestRouteStrategy());
            planner.ExecuteStrategy();
        }
    }
}
